cookbooks/cdo-varnish/test/integration/default/minitest/test_default.rb

```diff
@@ -0,0 +1,293 @@
+end
+
+# Mocks a simple text/plain response body at the specified URL.
+def mock_url(url, body, request_headers={}, response_headers={}, method='GET')
+  json = {
+  request: {
+    method: method,
+    url: url,
+    headers: Hash[*request_headers.map{|k, v|[k,{equalTo: v}]}.flatten]
+  },
+  response: {
+    status: 200,
+    body: body,
+    headers: {
+      "Content-Type" => "text/plain"
+    }.merge(response_headers)
+  }
+}.to_json
+  `curl -s -X POST #{LOCALHOST}:#{DASHBOARD_PORT}/__admin/mappings/new -d '#{json}'`
+end
+
+def _request(url, headers={}, cookies={}, method='GET')
+  header_string = headers.map { |key, value| "-H \"#{key}: #{value}\"" }.join(' ')
+  cookie_string = cookies.empty? ? '' : "--cookie \"#{cookies.map{|k,v|"#{k}=#{v}"}.join(';')}\""
+  `curl -X #{method} -s #{cookie_string} #{header_string} -i #{url}`.tap{assert_equal 0, $?.exitstatus}
+end
+
+def request(url, headers={}, cookies={})
+  _request("#{LOCALHOST}:#{DASHBOARD_PORT}#{url}", headers, cookies)
+end
+
+def proxy_request(url, headers={}, cookies={}, method='GET')
+  headers.merge!(host: '_default-studio.code.org')
+  headers.merge!('X-Forwarded-Proto' => 'https')
+  _request("#{LOCALHOST}:#{VARNISH_PORT}#{url}", headers, cookies, method)
+end
+
+def code(response)
+  /HTTP\/1\.1 (\d+)/.match(response.lines.to_a.first.strip)[1].to_i
+end
+
+def assert_ok(response)
+  assert_equal 200, code(response)
+end
+def assert_cache(response, hit)
+  assert_equal hit ? 'HIT' : 'MISS', /X-Varnish-Cache: (\w+)/.match(response)[1]
+end
+def assert_miss(response)
+  assert_cache response, false
+end
+def assert_hit(response)
+  assert_cache response, true
+end
+
+def last_line(response)
+  response.lines.to_a.last.strip
+end
+
+describe 'http' do
+  it 'handles a simple request' do
+    url = '/cache1'
+    text = 'Hello World!'
+    mock_url(url, text)
+    response = request url
+    assert_ok response
+    assert_equal text, last_line(response)
+  end
+
+  it 'caches a simple request' do
+    url = '/cache2'
+    text = 'Hello World 2!'
+    mock_url(url, text)
+
+    response = proxy_request url
+    assert_ok response
+    assert_miss response
+
+    response = proxy_request url
+    assert_ok response
+    assert_hit response
+
+    # Verify only one request hit the origin server
+    assert_equal 1, count_requests(url)
+  end
+
+  it 'redirects HTTP to HTTPS' do
+    response = _request "#{LOCALHOST}:#{VARNISH_PORT}/https"
+    assert_equal 301, code(response)
+    assert_equal "https://#{LOCALHOST}/https", /Location: ([^\s]+)/.match(response)[1]
+  end
+
+  it 'separately caches responses that vary on X-Varnish-Accept-Language' do
+    url = '/cache3'
+    text_en = 'Hello World!'
+    text_fr = 'Bonjour le Monde!'
+    mock_url(url, text_en, {'X-Varnish-Accept-Language' => 'en'}, {vary: 'X-Varnish-Accept-Language'})
+    mock_url(url, text_fr, {'X-Varnish-Accept-Language' => 'fr'}, {vary: 'X-Varnish-Accept-Language'})
```

Comment:
Would be curious to see if you can repro any bad behavior where Accept-Language = 'f' (i.e. less than two characters.) Odds of a crash are small of course.